src/SFA.DAS.ApprenticeCommitments/Data/IApprenticeshipContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SFA.DAS.ApprenticeCommitments.Data.Models;
+using SFA.DAS.ApprenticeCommitments.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace SFA.DAS.ApprenticeCommitments.Data
+{
+    public interface IApprenticeshipContext : IEntityContext<Apprenticeship>
+    {
+        internal async Task<List<Apprenticeship>> FindByApprenticeId(Guid apprenticeId)
+            => await Entities.Where(a => a.Apprentice.Id == apprenticeId).ToListAsync();
+
+        internal async Task<Apprenticeship> GetById(Guid apprenticeId, long apprenticeshipId)
+            => (await Find(apprenticeId, apprenticeshipId))
+                ?? throw new DomainException(
+                    $"Apprenticeship {apprenticeshipId} for {apprenticeId} not found");
+
+        public async Task<Apprenticeship?> Find(Guid apprenticeId, long apprenticeshipId)
+            => await Entities
+                .SingleOrDefaultAsync(
+                    a => a.Id == apprenticeshipId &&
+                    a.Apprentice.Id == apprenticeId);
+    }
+}
```

Comment:
Interesting new use of code in interfaces. I'm not absolutely sure about adding all our coding to interfaces. I can see good reasons why it should be done.... but I like it as a learning exercise

Reply:
The methods could be added as extension methods, I'm not sure I really see much difference between the two approaches.  Time will tell :)